src/web/client/context/userDataMap.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+export interface IUserData {
+    containerId: string;
+    fileName: string;
+    filePath: string;
+    userName: string;
+    userId: string;
+}
+
+export class UserData implements IUserData {
+    _containerId: string;
+    _fileName: string;
+    _filePath: string;
+    _userName: string;
+    _userId: string;
+
+    // Getters
+    public get containerId(): string {
+        return this._containerId;
+    }
+    public get fileName(): string {
+        return this.fileName;
+    }
+    public get filePath(): string {
+        return this.filePath;
+    }
+    public get userName(): string {
+        return this._userName;
+    }
+    public get userId(): string {
+        return this._userId;
+    }
+
+    constructor(
+        containerId: string,
+        fileName: string,
+        filePath: string,
+        userName: string,
+        userId: string
+    ) {
+        this._fileName = fileName;
+        this._containerId = containerId;
+        this._filePath = filePath;
+        this._userName = userName;
+        this._userId = userId;
+    }
+}
+
+export class UserDataMap {
+    private usersMap: Map<string, UserData> = new Map<string, UserData>();
+
+    public get getUserMap(): Map<string, UserData> {
+        return this.usersMap;
+    }
+
+    public setUserData(
+        containerId: string,
+        fileName: string,
+        filePath: string,
+        userName: string,
+        userId: string
+    ) {
+        const userData = new UserData(
+            containerId,
+            fileName,
+            filePath,
+            userName,
+            userId
+        );
+
+        this.usersMap.set(userId, userData);
```

Comment:
If we are treating userId as key  -we should consider splitting above interface and keeping an array in Map value. This can be done in follow-up PR